### DISCUSSION TITLE
Server not discoverable through the friends tab as a LAN world. Although connects when directly connecting using the IP.

### DISCUSSION THREAD

**Discussion by NanoQuark** [itzg/docker-minecraft-bedrock-server#591] [Q&A]
Using `tcpdump` I see that the server pings my other device on the same network when I punch in the address of the host (Raspberry Pi 4B). Connection works and I see no problems other than the need for my server to BE discoverable on LAN for easy cross-compatibility. However, I also noticed something weird. When removing the saved server in the servers tab, it then immediately appears in the LAN games section, then disappears on the next ping. I'm not sure if this is a bug or a network problem as I am using a public university network, locally only.

**Comment by itzg** [comment]
This image only downloads and runs the software provided by Mojang. Please contact their forum or issue tracker about issues specific to their software.

**Comment by ViperRNMC** [comment]
with version 1.21.30 the discovery works on my network

**Comment by kimonxx** [comment]
If you don't see the server in the friends tab as a LAN world, but can connect through the IP, check the _level.dat_ of your world. Look out for

> LANBroadcast:
> LANBroadcastIntent:

Both values must be set to 1 to be visible in the LAN.

_LANBroadcast: Whether the world has been opened with the "Visible to LAN players" world setting enabled.
LANBroadcastIntent: Whether the "Visible to LAN players" world toggle is enabled._

[https://minecraft.wiki/w/Bedrock_Edition_level_format#level.dat](url)

Took me couple of hours to solve my issue. After copying my world on my pc and openend it locally, and then copied it back to the server, the server (world) was not visible in the friends tab anymore, but I was able to connect directly through the servers tab. I didn't change the world settings of the LAN, no clue how that happened. After setting the value to 1 for LANBroadcast the world was visible again in the friends tab.

I have the same issue with the saved server in the servers tab not visible in the friends tab. Although my world stays visible in the friends tab, when I remove it from the servers tab. Maybe that's by design

**Reply by icnich** [reply]
> If you don't see the server in the friends tab as a LAN world, but can connect through the IP, check the _level.dat_ of your world. Look out for
> 
> > LANBroadcast:
> > LANBroadcastIntent:
> 
> Both values must be set to 1 to be visible in the LAN.
> 
> _LANBroadcast: Whether the world has been opened with the "Visible to LAN players" world setting enabled. LANBroadcastIntent: Whether the "Visible to LAN players" world toggle is enabled._
> 
> [https://minecraft.wiki/w/Bedrock_Edition_level_format#level.dat](url)
> 
> Took me couple of hours to solve my issue. After copying my world on my pc and openend it locally, and then copied it back to the server, the server (world) was not visible in the friends tab anymore, but I was able to connect directly through the servers tab. I didn't change the world settings of the LAN, no clue how that happened. After setting the value to 1 for LANBroadcast the world was visible again in the friends tab.
> 
> I have the same issue with the saved server in the servers tab not visible in the friends tab. Although my world stays visible in the friends tab, when I remove it from the servers tab. Maybe that's by design

@kimonxx 
Howdy - I am running into the same issues after doing exactly what you did. I moved my world from my server to my PC after backing up /world even the /server. Upon trying to add in marketplace add-ons in a single player game then move it back to my server, LAN discovery was broken but myself and friends abroad can still join via IP. I am trying to allow my kid to join via LAN on his Xbox still.

LANBroadcast in the .dat file and in the server.properties files are all showing LAN discovery enabled. Did you ever learn anything more about this? You're the first person I've seen that has described doing the same thing with the same result, but I've quadruple checked my values and they are all set to 1/true so your solution isn't working for me.

Thank you!
